instrumentation/internal/sqltestutil/doc.go

```diff
@@ -0,0 +1,22 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Package sqltestutil contains helpers for testing database instrumentations
+build on top of the splunksql instrumentation.
+This package is in a separate module from the instrumentation it tests to
+isolate the dependency of the default SDK and not impose this as a transitive
+dependency for users.
```

Comment:
This package doesn't seem to make sense. The `github.com/signalfx/splunk-otel-go/instrumentation/database/sql/splunksql/test` and `github.com/signalfx/splunk-otel-go/instrumentation/github.com/go-sql-driver/mysql/splunkmysql/test
` packages are already their own modules with isolated dependencies. By sharing this internal module across module boundaries it effectively makes this API public. This is something already done with the config, but I'm not sure we want to continue adding to this. Especially if we are not sure about the design of the API yet (give it will have to be a stable API).

Reply:
> give it will have to be a stable API.

It will not have to be as it is only used in `test` modules. The `instrumentation/internal` module has to be stable. But ``instrumentation/internal/sqltestutil` would not have to be if we use it only in `test` modules.

Reply:
https://github.com/signalfx/splunk-otel-go/pull/1973/commits/f359c10e63ba2527a92ef0ea6346665311682e45